CMakeLists.txt

```diff
@@ -405,6 +427,7 @@ SET(LIB_SOURCE_FILES
     lib/common/balancer/roundrobin.c
     lib/common/balancer/least_conn.c
     lib/common/absprio.c
+    lib/common/busypoll.c
```

Comment:
Rather than adding busypoll.c always, I think we might want to follow the precedence of adding feature-specific files only when that feature is available, as well as passing the feature flags using `#define`.

Using mruby support as an example, we do something like below.

In CMakeFiles.txt:
* set `WITH_MRUBY_DEFAULT` based on if dependencies are available,
* set `WITH_MRUBY` based on user input, or adopt `WITH_MRUBY_DEFAULT` if there is no user input, and
* if `WITH_MRUBY` is set, set `-DH2O_USE_MRUBY=1` as the compile flag and add add necessary source files as dependencies.

In the source files, use `#if H2O_USE_MRUBY` to see if the feature is available.

Reply:
PS. Maybe a better example is the diff of CMakeLists.txt in https://github.com/h2o/h2o/pull/3201/files.